compiler/rustc_mir_build/src/thir/pattern/_match.rs

```diff
@@ -2207,19 +2111,7 @@ fn pat_constructor<'tcx>(
             if let Some(int_range) = IntRange::from_const(tcx, param_env, value, pat.span) {
```

Comment:
Longer-term, would it make sense to make `PatKind::Constant` fully opaque, which I guess means not using it for integer ranges either?

Reply:
that won't work, as we have no way to distinguish literal integers in patterns from constants.

Reply:
I thought structural match things like integers end up with a different `PatKind` that `Constant`?

Reply:
They do not (yet). That is an independent refactoring that we should analyze in isolation on its effect on readability, diagnostics and perf.

Reply:
Ah I see.